src/test/java/be/syntra/devshop/DevshopBack/services/UserRoleServiceTest.java

```diff
@@ -0,0 +1,74 @@
+package be.syntra.devshop.DevshopBack.services;
+
+import be.syntra.devshop.DevshopBack.exceptions.UserRoleNotFoundException;
+import be.syntra.devshop.DevshopBack.security.entities.UserRole;
+import be.syntra.devshop.DevshopBack.security.repositories.UserRoleRepository;
+import be.syntra.devshop.DevshopBack.security.services.UserRoleService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+
+import java.util.List;
+import java.util.Optional;
+
+import static be.syntra.devshop.DevshopBack.security.entities.UserRoles.ROLE_ADMIN;
+import static be.syntra.devshop.DevshopBack.security.entities.UserRoles.ROLE_USER;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.*;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+public class UserRoleServiceTest {
+
+    @Mock
+    private UserRoleRepository userRoleRepository;
+
+    @InjectMocks
+    private UserRoleService userRoleService;
+
+    @BeforeEach
+    public void init() {
+        initMocks(this);
+    }
```

Comment:
this can be replaced with a @Before .... BUT i didn't succeed in that in my own tests because it always wanted import some other class in stead of the one i needed because fuck knows.  And this @Before that i'm  referring to is from a testing package not the AOP that intellij seems to love importing.

Reply:
conflict of dependencies from our POM. JUnit4 and 5 differ in @Before's. 5 allows for a @BeforeAll and a @BeforeEach. The first one needs to be static, which offers a conflict with our initMocks. The second one works. The @Before annotation demands a value and is indeed from the wrong import.